File: Samples/MediaPlayerCS/FloatToDoubleConverter.cs
﻿using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;
using Windows.UI.Xaml.Data;

namespace MediaPlayerCS
{
    public class FloatToDoubleConverter : IValueConverter
    {
        public object Convert(object value, Type targetType, object parameter, string language)
        {
            return (double)(float)value;
        }

        public object ConvertBack(object value, Type targetType, object parameter, string language)
        {
            return (float)(double)value;
        }
    }
}

File: Samples/MediaPlayerCPP/FloatToDoubleConverter.cpp
#include "pch.h"
#include "FloatToDoubleConverter.h"

Platform::Object^ MediaPlayerCPP::FloatToDoubleConverter::Convert(Platform::Object^ value, Windows::UI::Xaml::Interop::TypeName targetType, Platform::Object^ parameter, Platform::String^ language)
{
	auto box = static_cast<Platform::IBox<float>^>(value);
	return ref new Platform::Box<double>(box->Value);
}

Platform::Object^ MediaPlayerCPP::FloatToDoubleConverter::ConvertBack(Platform::Object^ value, Windows::UI::Xaml::Interop::TypeName targetType, Platform::Object^ parameter, Platform::String^ language)
{
	auto box = static_cast<Platform::IBox<double>^>(value);
	return ref new Platform::Box<float>((float)box->Value);
}
